Makefile

```diff
@@ -14,16 +14,24 @@ SHELLCHECK_CMD := shellcheck -e SC1091 -e SC2002 scripts/*
 # -e SC1091 exempts `source relative/path/to/file` errors
 # -e SC2002 exempts `Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.``
 TEST_ENV_PREFIX := docker run --rm -v ${CURDIR}:/bash -w /bash quay.io/deis/shell-dev
-
-build: docker-build
-push: docker-push
-run:
-	docker run -e AUTH_TOKEN="${AUTH_TOKEN}" \
+RUN_PREFIX := docker run -e AUTH_TOKEN="${AUTH_TOKEN}" \
 		-e CLI_VERSION="${CLI_VERSION}" \
 		-e CLUSTER_DURATION="${CLUSTER_DURATION}" \
 		-e CLUSTER_REGEX="${CLUSTER_REGEX}" \
 		-e CLUSTER_VERSION="${CLUSTER_VERSION}" \
-		-v "${E2E_DIR_LOGS}":/home/jenkins/logs:rw ${IMAGE}
+		-e JOB_NAME="${JOB_NAME}" \
+		-e BUILD_NUMBER="${BUILD_NUMBER}" \
+		-v "${E2E_DIR_LOGS}":/home/jenkins/logs:rw
+
+build: docker-build
+push: docker-push
+run:
+	${RUN_PREFIX} ${IMAGE}
+run-upgrade:
+	${RUN_PREFIX} \
+		-e RUN_E2E="${RUN_E2E}" \
+		-e ORIGIN_WORKFLOW_REPO="${ORIGIN_WORKFLOW_REPO}" \
+		-e UPGRADE_WORKFLOW_REPO="${UPGRADE_WORKFLOW_REPO}" ${IMAGE} upgrade
```

Comment:
@mboersma I think it's needed here (the reason for the ENTRYPOINT). We could change this to `run.sh upgrade` instead and revert to the CMD argument.

Typically when an ENTRYPOINT is defined in a Dockerfile, it's to force users into running some form of a bootstrap script before starting the image. For example, [in postgres's case](https://github.com/deis/postgres/blob/master/rootfs/docker-entrypoint.sh) if the user runs `docker run deis/postgres postgres` then it spawns postgres, but if they run `docker run -it deis/postgres bash` then that works normally as expected. Similarly, [dind (or docker-in-docker)](https://github.com/jpetazzo/dind) has a wrapper script called [wrapdocker](https://github.com/jpetazzo/dind/blob/master/wrapdocker) that is only invoked when `docker run dind` is invoked, allowing the user to bypass the wrapper if they do not wish to run the docker daemon (such as when one is debugging the image itself).

Perhaps the run script could be re-written so that it follows that mantra, but I imagine that's too much work for this PR. WDYT, @vdice?